modules/gapi/src/streaming/onevpl/engine/decode/decode_engine_legacy.cpp

```diff
@@ -0,0 +1,310 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <algorithm>
+#include <exception>
+
+#include <opencv2/gapi/streaming/onevpl/data_provider_interface.hpp>
+
+#include "streaming/onevpl/engine/decode/decode_engine_legacy.hpp"
+#include "streaming/onevpl/engine/decode/decode_session.hpp"
+#include "streaming/onevpl/accelerators/accel_policy_interface.hpp"
+#include "streaming/onevpl/accelerators/surface/surface.hpp"
+#include "streaming/onevpl/utils.hpp"
+#include "logger.hpp"
+
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+/* UTILS */
+mfxU32 GetSurfaceSize(mfxU32 FourCC, mfxU32 width, mfxU32 height) {
+    mfxU32 nbytes = 0;
+
+    mfxU32 half_width = width / 2;
+    mfxU32 half_height = height / 2;
+    switch (FourCC) {
+        case MFX_FOURCC_I420:
+        case MFX_FOURCC_NV12:
+            nbytes = width * height +  2 * half_width * half_height;
+            break;
+        case MFX_FOURCC_I010:
+        case MFX_FOURCC_P010:
+            nbytes = width * height + 2 * half_width * half_height;
```

Comment:
Should this size in bytes include alignment?

In some platforms, image data (stride) is aligned to some multiplier and may exceed its logical size.

Reply:
not sure, unfortunately. usually we learn about this limitations when enabling this or that particular platform. even more, on kmb one revision had one alignment and other revision has required the other.

At least I'd make it configurable somehow